linux/src/threadpool_linux.c

```diff
@@ -49,34 +50,31 @@
 MU_DEFINE_ENUM(TASK_RESULT, TASK_RESULT_VALUES);
 MU_DEFINE_ENUM_STRINGS(TASK_RESULT, TASK_RESULT_VALUES)
 
-#define TIMER_GUARD_VALUES \
-    OK_TO_WORK,            \
-    TIMER_WORKING,         \
-    TIMER_DELETING
-
-MU_DEFINE_ENUM(TIMER_GUARD, TIMER_GUARD_VALUES);
-MU_DEFINE_ENUM_STRINGS(TIMER_GUARD, TIMER_GUARD_VALUES);
-
-typedef struct TIMER_INSTANCE_TAG
+typedef struct TIMER_TAG
 {
     THREADPOOL_WORK_FUNCTION work_function;
     void* work_function_ctx;
     timer_t time_id;
-    INTERLOCKED_DEFINE_VOLATILE_STATE_ENUM(TIMER_GUARD, timer_work_guard);
-} TIMER_INSTANCE;
+    SRW_LOCK_LL timer_lock;
+} TIMER;
+
+THANDLE_TYPE_DEFINE(TIMER);
 
 typedef struct THREADPOOL_TASK_TAG
 {
     volatile_atomic int32_t task_state;
     THREADPOOL_WORK_FUNCTION work_function;
     void* work_function_ctx;
+    volatile uint32_t pending_api_calls;
+    volatile_atomic int32_t *pending_work_item_count_ptr;
 } THREADPOOL_TASK;
```

Comment:
I think this was removed by Nishikant also.

Reply:
fixed